API/Services/DataRetrievalService.cs

```diff
@@ -0,0 +1,41 @@
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Http.Json;
+using System.Threading.Tasks;
+using Backend.API.Data;
+
+namespace Backend.API.Services
+{
+    public interface IDataRetrievalService
+    {
+        Task<Todo> GetTodo(int todoId);
+
+        Task<Todo[]> GetTodos();
+    }
+
+    public class DataRetrievalService : IDataRetrievalService
+    {
+        private HttpClient HttpClient { get; }
+
+        public DataRetrievalService()
+        {
+            HttpClient = new HttpClient();
+            HttpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+            HttpClient.DefaultRequestHeaders.UserAgent.ParseAdd(
+                "Mozilla/5.0 (Macintosh; Intel Mac OS X x.y; rv:42.0) Gecko/20100101 Firefox/42.0");
```

Comment:
> All requests must (if possible) include an identifying User Agent-string (UA) in the request with the application/domain name, optionally version number. You should also include a company email address or a link to the company website where we can find contact information. If we cannot contact you in case of problems, you risk being blocked without warning.

The UserAgent should be updated to reflect this.